openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1507,7 +1507,96 @@ It is up to the Credential Issuer whether to update both the signature and the c
 
 # Privacy Considerations
 
-TBD
+When [@!RFC9396] is used, the Privacy Considerations of that specification also apply.
+
+The privacy principles of [@ISO.29100] should be adhered to.
+
+## User Consent
+
+The Credential Issuer should obtain the End-User's consent before issuing Credential(s)
+to the Wallet. It should be made clear to the End-User what information is being included in the
+Credential(s) and for what purpose.
+
+## Minimum Disclosure
+
+To ensure minimum disclosure and prevent Verifiers from obtaining claims unnecessary for the
+transaction at hand, when issuing Credentials that are intended to be created once and then used a
+number of times by the End-User, the Credential Issuers and the Wallets SHOULD implement
+Credential formats that support selective disclosure, or consider issuing a separate Credential
+for each user claim.
+
+## Storage of the Credentials
+
+To prevent a leak of End-User data, especially when it is signed, which risks revealing private data of End-Users to
+third parties, systems implementing this specification SHOULD be designed to minimize the amount
+of End-User data that is stored. All involved parties SHOULD store Verifiable Credentials
+containing privacy-sensitive data only for as long as needed, including in log files. Any logging of
+End-User data should be carefully considered as to whether it is necessary at all. The time logs are retained
+for should be minimized.
+
+After Issuance, Credential Issuers SHOULD NOT store the Issuer-signed Credentials if they
+contain privacy-sensitive data. Wallets SHOULD store Credentials only in encrypted form, and,
+wherever possible, use hardware-backed encryption. Credentials that are no longer valid to present should be deleted as soon as
+possible.
```

Comment:
```suggestion
wherever possible, use hardware-backed encryption.
```

This might be undesirable. I imagine there may be use cases where presenting an expired credential may still be helpful, for example presenting an expired credential to the issuer as an authentication factor for provisioning a fresh credential or simply as a convenient way to share data about oneself (without the guarantee of the validity of the data).

IMO, it is a policy decision of the relying party whether or not they wish to accept expired credentials, and the credential holder should have ultimate control over their credentials. This suggestion would be equivalent to a physical ID card (safely) incinerating itself upon reaching the expiry date.

Reply:
> The most important rule to keep in mind is that anything must be under the user's control, even the deletion.

I don't think this is anything we should be including in the VCI spec; there are cases (e.g. fraud) where it would be completely valid for the wallet to purge a credential without the user agreeing to it.

Reply:
text ` Credentials that are no longer valid to present should be deleted as soon as possible.` has been deleted already, so I think your comment has been addressed, @cobward ?

Reply:
@Sakurann That text is still there. (We did remove the mention of 'expired' after last week's WG call but kept the text, I think Jacob is the first person I'm aware of that has suggested removed that text entirely.)

Reply:
Thanks all for the clarification, and I'm sorry that I missed this context from the WG calls; they almost always clash with other immovable meetings for me. Are there minutes for these calls that I can use to keep up with the conversations?

Back on the topic:
> The word 'expired' was deliberately avoided for exactly the reasons you say

Ah right. For this reason, I think the fact that I read it this way could mean that the text is ambiguous and at the very least needs to clarify what `valid to present` actually means.

> I would expect ecosystems and/or issuers will probably provide a definition for 'valid to present'.

If this is what we mean then we should be specific, and say something to the effect of:
_Credentials that are **deemed** no longer valid to present **by the given credential ecosystem** should be deleted as soon as possible._

I still think it should be up to the user to decide whether or not their credential gets deleted, as they may have uses for the credential that the wallet developer is not aware of. I don't think it's necessarily true that credentials always exist in closed and unchanging ecosystems.

>  there are cases (e.g. fraud)

We're talking about validity here though, not fraud. Would you have the same problem with this modification of Giuseppe's sentence? _The most important rule to keep in mind is that anything must be under **a legitimate** user's control, even the deletion._

Reply:
my UX is that nobody can put his hands in my pocket, open my wallet, and remove stuffs. As in the real world, as in the digital one :-)

Reply:
>  Are there minutes for these calls that I can use to keep up with the conversations?

Yes, they get sent out to the mailing list - you can subscribe or view the archives here: https://lists.openid.net/mailman/listinfo/openid-specs-digital-credentials-protocols

> Credentials that are deemed no longer valid to present by the given credential ecosystem should be deleted as soon as possible.

I think this presupposes ecosystems etc which aren't currently ever mentioned in the VCI spec; these are privacy recommendations so should be kept general.

> We're talking about validity here though, not fraud.

I think they're related though; my presumption would be that a credential that is discovered to have been fraudulently obtained would likely become invalid to present. (Is it the word 'valid' that's causing confusion - if so any suggestions for alternative words are welcome!)

> The most important rule to keep in mind is that anything must be under a legitimate user's control, even the deletion.

I do appreciate the comparison removing a physical credential from a user's physical wallet but I am not convinced that a Wallet/Issuer should need to obtain the legitimate user's permission to delete a credential that subsequently turns out to have been fraudulently issued.

The discussion on the WG call did seek to draw a line between 'credential content has expired' (so a physical passport that has reached it's "valid until" date is potentially no longer valid for travel but might still be valid for identity proofing)  with (e.g. in the case of an SD-JWT) 'credential format cryptographic wrapping `exp` value' - the consensus on the call seemed to be towards the latter being something that was generally expected to be honoured, I'm interested to know if there is actually consensus on that.

Reply:
> 
> While validity and fraud are similar we should explicitly talk about each separately rather than conflating them into a single problem, since they should have different outcomes.
> 

I agree. For clarity it should be addressed in a separate line or paragraph. 

I also wanted to point out that in the same paragraph we have the first line that defines the intended behavior of the Issuer, the second that explicitly refers to the Wallet and the third one without an explicit subject. I would explictly mention a Wallet there for clarity.

>Credentials that are deemed no longer valid to present by all concerned parties (the holder and the verifiers >that the holder can reasonably be expected to interact with) SHOULD be deleted as soon as possible.

When it comes to the above sentence, it became a little verbose. In addition to that it should take into account UX which was raised by @peppelinux too. 
>SHOULD be deleted as soon as possible
The deletion does not have to happen automagically. I imagine that it could be indicated to a user in some way that such credentials are in their wallet, suggest deletion and let them do it.

Reply:
> Credentials that are deemed no longer valid to present by all concerned parties (the holder and the verifiers that the holder can reasonably be expected to interact with) SHOULD be deleted as soon as possible.

I am not sure this is very actionable. it assumes there is always an agreement between verifier and wallet regarding whether the wallet can present an "expired" credential.

current sentence `no longer valid to be presented` implicitly includes making the determination what is `valid` which I think was the main point? we could also consider changing to `Wallet should delete Credentials that are no longer _verifiable_ to be presented as soon as possible.`

Reply:
based on the WG call, we changed the language to what is in the thread below: https://github.com/openid/OpenID4VCI/pull/244/files#r1480442717
moving away from valid/verifiable terms

Reply:
> I am not sure this is very actionable. it assumes there is always an agreement between verifier and wallet regarding whether the wallet can present an "expired" credential.
> 
> current sentence no longer valid to be presented implicitly includes making the determination what is valid which I think was the main point?

Does that imply that it is up to the wallet alone to determine the validity of the credential?

> we could also consider changing to Wallet should delete Credentials that are no longer _verifiable_ to be presented as soon as possible.

Without a definition of "verifiable" I don't think it makes much difference. For example, if I can verify the signature, is that enough to be considered verifiable?

I think we need to take one of two paths:
1. Include or reference a definition of "valid to present"/"verifiable".
2. Explicitly state which party(s) can determine what "valid to present"/"verifiable" means.

Reply:
Ah sorry @Sakurann, not sure why I didn't see your comment when I responded. That text works for me, so will resolve this thread.